payments-ui-core/src/main/java/com/stripe/android/ui/core/elements/CardNumberController.kt

```diff
@@ -1,18 +1,43 @@
 package com.stripe.android.ui.core.elements
 
+import android.content.Context
+import androidx.annotation.VisibleForTesting
 import androidx.compose.ui.text.input.KeyboardCapitalization
 import androidx.compose.ui.text.input.KeyboardType
+import com.stripe.android.cards.CardAccountRangeRepository
+import com.stripe.android.cards.CardAccountRangeService
+import com.stripe.android.cards.CardNumber
+import com.stripe.android.cards.DefaultCardAccountRangeRepositoryFactory
+import com.stripe.android.cards.DefaultStaticCardAccountRanges
+import com.stripe.android.cards.StaticCardAccountRanges
+import com.stripe.android.model.AccountRange
 import com.stripe.android.model.CardBrand
 import com.stripe.android.ui.core.forms.FormFieldEntry
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.combine
 import kotlinx.coroutines.flow.map
+import kotlin.coroutines.CoroutineContext
 
 internal class CardNumberController constructor(
     private val cardTextFieldConfig: CardNumberConfig,
+    private val cardAccountRangeRepository: CardAccountRangeRepository,
+    private val workContext: CoroutineContext,
+    private val staticCardAccountRanges: StaticCardAccountRanges = DefaultStaticCardAccountRanges(),
```

Comment:
I think we can move StaticCardAccountRanges into the CardAccountRangeService.  It feels like the CardNumberController shouldn't care about if it needs to query the StaticCardAccount or the server to get the AccountResult.   All it needs to know is to call update (a new function) on CardAccountRangeService, and it will get updated on any required changes to the AccountResult through the callback.

Reply:
There's a couple places where we're passing in a custom staticCardAccountRanges into CardNumberEditText instead of using the default, how would I handle the custom passing into the CardAccountRangeService?